lib/diagnostic.js

```diff
@@ -6,26 +6,38 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const normalizeWords = function (stringOfWords) {
+  return stringOfWords.replace(/\W+/,'').toUpperCase();
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const uniqueWords = function (stringOfWords) {
+  let result = {};
+
+  stringOfWords.forEach(function (word) {
+    result[word] = 0;
+  });
+
+  return result;
 };
 
 // 3. Write a function that returns the count of words in a string.
 // Provid the *option* to count unique words instead of total words.
-const wordCount = function (/* */) {
-  /* your response here */
+const wordCount = function () {
+
```

Comment:
It would be something like 

```javascript
const wordCount = function (text, unique) {
  if (unique) {
    return uniqueWords(text).length;
  }

  return normalizeWords(text).length;
};
```